src/prolly/map.rs

```diff
@@ -116,6 +125,75 @@ impl Map {
         self.pending.clear();
         Ok(self.base.as_ref().unwrap().chunk().hash().into())
     }
+
+    // Returns the diff between the pending entries and the already flushed entries.
+    pub fn pending_diff(&self) -> Result<Vec<String>, FromUtf8Error> {
+        let mut keys = Vec::with_capacity(self.pending.len());
+        for (key, pending_val) in self.pending.iter() {
+            match pending_val {
+                Some(pending_val) => match self.base_get(key) {
+                    Some(base_val) => {
+                        if pending_val != base_val {
+                            keys.push(String::from_utf8(key.clone())?);
+                        }
+                    }
+                    None => {
+                        keys.push(String::from_utf8(key.clone())?);
+                    }
+                },
+                None => {
+                    if self.base_has(key) {
+                        keys.push(String::from_utf8(key.clone())?);
+                    }
+                }
+            }
+        }
+        Ok(keys)
+    }
+
+    /// Returns the keys that are different between two maps.
+    pub fn diff<'a>(a: &'a Self, b: &'a Self) -> Result<Vec<String>, FromUtf8Error> {
```

Comment:
One optimization I didn't do is to create key ranges of this... we can do that later.